Allow for use in other Packages

spprichard opened the pull request:
- Added library product to allow other packages to use this code

spprichard committed:
Added importable library product

spprichard committed:
set min iOS version

jkrukowski commented on Package.swift:
```diff
@@ -5,11 +5,18 @@ import PackageDescription
             name: "SQLiteVecCLI",
             targets: ["SQLiteVecCLI"]
         ),
+        .library(
+            name: "SQLiteVec",
+            targets: ["SQLiteVec"]
+        ),
```
good call, could you please add as well

```swift
.library(
    name: "CSQLiteVec",
    targets: ["CSQLiteVec"]
),
```

spprichard replied:
Sure no problem 👍

jkrukowski commented on Package.swift:
```diff
@@ -5,11 +5,18 @@ import PackageDescription
 
 let package = Package(
     name: "SQLiteVec",
+    platforms: [
+        .iOS(.v16)
+    ],
```
I wouldn't add it. Specifying `platforms` (according to the [docs](https://github.com/swiftlang/swift-package-manager/blob/main/Documentation/PackageDescription.md#supportedplatform)) means

> By default, the Swift Package Manager assigns a predefined minimum deployment version for each supported platforms unless you configure supported platforms using the platforms API. This predefined deployment version is the oldest deployment target version that the installed SDK supports for a given platform. 

So it should work on iOS without explicitly specifying it here

spprichard replied:
Ya I was curious about that. Yup not problem, I'll remove

jkrukowski reviewed (commented):
Thanks for the contribution! Left some comments, lmk what you think

spprichard committed:
PR Comment Response

- Min iOS version doens't need to be explicitly set
- Add CSQLiteVec as an importable product
